src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -1413,7 +1416,112 @@ private IBlockOperation CreateBoundBlockOperation(BoundBlock boundBlock)
+
+            var argumentNullExceptionMethodSymbol = (IMethodSymbol)compilation.GetWellKnownTypeMember(WellKnownMember.System_ArgumentNullException__ctorString);
+            var argumentNullExceptionType = compilation.GetWellKnownType(WellKnownType.System_ArgumentNullException);
+            var argumentNullExceptionObject = new ObjectCreationOperation(
+                        constructor: argumentNullExceptionMethodSymbol,
+                        initializer: new ObjectOrCollectionInitializerOperation(
+                                initializers: ImmutableArray<IOperation>.Empty,
+                                _semanticModel,
+                                syntax,
+                                compilation.GetWellKnownType(WellKnownType.System_ArgumentNullException),
+                                constantValue,
+                                isImplicit: true),
+                        ImmutableArray.Create<IArgumentOperation>(
+                            new ArgumentOperation(
+                                new LiteralOperation(_semanticModel, syntax, compilation.GetSpecialType(SpecialType.System_String), parameter.Name, isImplicit: true),
+                                ArgumentKind.Explicit,
+                                parameter: argumentNullExceptionMethodSymbol.Parameters[0],
+                                inConversionOpt: null,
+                                outConversionOpt: null,
+                                _semanticModel,
+                                syntax,
+                                isImplicit: true)),
+                        _semanticModel,
+                        syntax,
+                        type: argumentNullExceptionType,
+                        constantValue,
+                        isImplicit: true);
+
+            IOperation whenTrue = new ExpressionStatementOperation(
+                new ThrowOperation(
+                    argumentNullExceptionObject,
+                    _semanticModel,
+                    syntax,
+                    type: argumentNullExceptionType,
+                    constantValue,
+                    isImplicit: true),
+                _semanticModel,
+                syntax,
+                compilation.GetSpecialType(SpecialType.System_Void),
+                constantValue,
+                isImplicit: true);
+            return new ConditionalOperation(conditionOp, whenTrue, null, isRef: false, _semanticModel, syntax, compilation.GetSpecialType(SpecialType.System_Boolean), constantValue, isImplicit: true);
+
+        }
+
+        private ImmutableArray<ConditionalOperation> GenerateNullChecksForParameters(ImmutableArray<ParameterSymbol> parameters, SyntaxNode syntax)
+        {
+            ArrayBuilder<ConditionalOperation> ifList = null;
+            foreach (var param in parameters)
+            {
+                if (param.IsNullChecked)
+                {
+                    ifList ??= ArrayBuilder<ConditionalOperation>.GetInstance();
```

Comment:
>GetInstance() [](start = 66, length = 13)

Providing the initial capacity is a perf boost, we generally do this when we can to avoid reallocating internal arrays.

Reply:
We don't know how many parameters are null-checked. It seems cleaner to leave the size unspecified.

---
In reply to: [305548807](https://github.com/dotnet/roslyn/pull/37381#discussion_r305548807) [](ancestors = 305548807)